verifiers/internal/gha/provenance_test.go

```diff
@@ -564,7 +574,7 @@ func Test_VerifyWorkflowInputs(t *testing.T) {
 				"some_bool":       "true",
 				"some_integer":    "123",
 			},
-			expected: serrors.ErrorInvalidDssePayload,
+			expected: serrors.ErrorMismatchWorkflowInputs,
```

Comment:
Reviewers: I had to update this testcase and the error it generated because it was actually a weirdly formed DSSE envelope input (doubly encoded), and my code made it fail when it was detecting predicateType. Meanwhile, the error had to change from InvalidDssePayload to the actual error check, which is mismatch workflow inputs.